src/main/kotlin/com/sourcegraph/cody/initialization/EndOfTrialNotificationScheduler.kt

```diff
@@ -0,0 +1,103 @@
+package com.sourcegraph.cody.initialization
+
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.CurrentUserCodySubscription
+import com.sourcegraph.cody.agent.protocol.GetFeatureFlag
+import com.sourcegraph.cody.agent.protocol.Plan
+import com.sourcegraph.cody.agent.protocol.Status
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.config.ConfigUtil
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+class EndOfTrialNotificationScheduler private constructor(val project: Project) : Disposable {
+
+  private val logger = Logger.getInstance(EndOfTrialNotificationScheduler::class.java)
+
+  private val scheduler = Executors.newScheduledThreadPool(1)
+
+  init {
+    scheduler.scheduleAtFixedRate(
+        /* command = */ {
+          if (!ConfigUtil.isCodyEnabled()) {
+            return@scheduleAtFixedRate
+          }
+
+          if (project.isDisposed) {
+            this.dispose()
+          }
+
+          if (CodyAuthenticationManager.instance.getActiveAccount(project)?.isDotcomAccount() ==
+              false) {
+            return@scheduleAtFixedRate
+          }
+
+          CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
+            agent.server
+                .getCurrentUserCodySubscription()
+                .thenApply { currentUserCodySubscription ->
```

Comment:
DO NOT run `thenApply` on server responses.
Or, run it if you want, but first answer me this:
on which thread/threadpool this will be running, and how big that thread pool is? :)
I do not consider it to be safe, maybe we should so some automagical wrapping just to be able to use it without thinking about it.

Reply:
I do `completeOnTimeout` in the same block in line 63. I think it will be running on the very same thread (started by `applyAgentOnBackgroundThread` in line 39). 

I do not understand your concerns in this case but likely I miss something. Let's discuss it via f2f.

Reply:
How it can be running in the same thread, if the current thread can even finish in the meantime? :) This is not a blocking operation. It's tricky, I agree, let's discuss on the standup tomorrow.

Reply:
I refactored the code so it's easier to read and understand.

Reply:
Agree, it's much more readable now, thank you!